### PR TITLE
Adds a check to error out if the run command didn't start properly.

### LINKED ISSUE #4197: odo push does not fail if executed devfile command failed to start
/kind bug

## What versions of software are you using?

**Operating System:** Linux

**Output of `odo version`:** 2.0.1

## How did you run odo exactly?
```
user@host:~$ git clone https://github.com/dgolovin/nodejs-ex ./nodejs-devfile-issue
Cloning into './nodejs-devfile-issue'...
remote: Enumerating objects: 637, done.
remote: Total 637 (delta 0), reused 0 (delta 0), pack-reused 637
Receiving objects: 100% (637/637), 253.59 KiB | 2.59 MiB/s, done.
Resolving deltas: 100% (246/246), done.

user@host:~$ cd nodejs-devfile-issue/

user@host:~/nodejs-devfile-issue$ odo create nodejs --app app1 --project p1
Validation
 ✓  Checking devfile existence [35224ns]
 ✓  Checking devfile compatibility [55663ns]
 ✓  Creating a devfile component from registry: DefaultDevfileRegistry [55176ns]
 ✓  Validating devfile component [94611ns]

Please use `odo push` command to create the component with source deployed
```
## Actual behavior

```
user@host:~/nodejs-devfile-issue$  odo push --debug

Validation
 ✓  Validating the devfile [59256ns]

Creating Kubernetes resources for component nodejs
 ✓  Waiting for component to start [6s]

Applying URL changes
 ✓  URL http-3000: http://http-3000-nodejs-p1.apps-crc.testing/ created

Syncing to component nodejs
 ✓  Checking files for pushing [677080ns]
 ✓  Syncing files to the component [223ms]

Executing devfile commands for component nodejs
 ✓  Executing install command "npm install" [5s]
 ✓  Executing debug command "npm run debug" [1s]

Pushing devfile component nodejs
 ✓  Changes successfully pushed to component

user@host:~/nodejs-devfile-issue$ odo log
time="2020-11-05T09:06:56Z" level=info msg="create process:devrun" 
time="2020-11-05T09:06:56Z" level=info msg="create process:debugrun" 
time="2020-11-05T09:07:02Z" level=debug msg="no auth required" 
time="2020-11-05T09:07:02Z" level=debug msg="succeed to find process:debugrun" 
time="2020-11-05T09:07:02Z" level=info msg="try to start program" program=debugrun 
time="2020-11-05T09:07:02Z" level=info msg="success to start program" program=debugrun 
ODO_COMMAND_DEBUG is npm run debug
Changing directory to /project
Executing command cd /project && npm run debug
npm ERR! missing script: debug

npm ERR! A complete log of this run can be found in:
npm ERR!     /opt/app-root/src/.npm/_logs/2020-11-05T09_07_03_073Z-debug.log
time="2020-11-05T09:07:03Z" level=debug msg="wait program exit" program=debugrun 
time="2020-11-05T09:07:03Z" level=error msg="program stopped with error:exit status 1" program=debugrun 
time="2020-11-05T09:07:03Z" level=info msg="Don't start the stopped program because its retry times 0 is greater than start retries 0" program=debugrun
```

Odo push does not fails even if executed devfile command does.

## Expected behavior

If executing devfile command fails, odo push fails.

### PR DESCRIPTION
**Pull request by mik-dass**
**What type of PR is this?**

/kind bug

**What does does this PR do / why we need it**:

It waits for 1 sec to check the status and displays the last 20 lines in case of a failure.

**Which issue(s) this PR fixes**:

Fixes #4197

**PR acceptance criteria**:

- [ ] Unit test 

- [X] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

- Push a component with a invalid run command, it should error out with the last 20 lines of the component's log.
- Repeat the above steps with a invalid debug command and using the `debug` flag during push.

### CODE REVIEW

**Comment by girishramnani** [comment]
unit test failing

File: pkg/devfile/adapters/kubernetes/component/adapter.go [hunk 1]
```diff
@@ -230,6 +234,20 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
...
 			return err
 		}
+
+		runCommand := pushDevfileCommands[devfilev1.RunCommandGroupKind]
+		if parameters.Debug {
+			runCommand = pushDevfileCommands[devfilev1.DebugCommandGroupKind]
+		}
+
+		// wait for a second
+		wait := time.After(supervisorDStatusWaitTimeInterval * time.Second)
```
Comment (girishramnani) [inline pkg/devfile/adapters/kubernetes/component/adapter.go:244]: why wait here?

Comment (mik-dass) [reply pkg/devfile/adapters/kubernetes/component/adapter.go:244]: It's a requirement mentioned in https://github.com/openshift/odo/issues/4197#issuecomment-724113853

**Review by dharmit — APPROVED** [review]
/lgtm

**Pull request merged**